conda-store-server/environment-dev.yaml

```diff
@@ -1,63 +1,31 @@
+# only development dependencies should be added to this file, if you add a new dev dependency
+# you must ensure to add it to the [tool.hatch.envs.dev] section of the pyproject.toml file too
+# required conda-store-server or conda-store (core) dependencies MUST ONLY be added/updated in the [dependencies]
+# section of the corresponding pyproject.toml file
+# the only exceptions to this rule are conda and constructor which are only conda installable
 name: conda-store-server-dev
 channels:
   - conda-forge
-  - microsoft
   - nodefaults
 dependencies:
   - python ==3.10
   # conda builds
   - conda ==23.5.2
```

Comment:
Why this particular pin?

Reply:
Seems this was introduced in https://github.com/conda-incubator/conda-store/pull/522 -> going through the commits this seems was temporary while https://github.com/mamba-org/mamba/issues/2715

Will remove it and see how it goes